slamd/common/session_controller.py

```diff
@@ -0,0 +1,48 @@
+from datetime import datetime
+
+from flask import Blueprint, make_response, request, redirect
+
+from slamd.common.error_handling import SlamdUnprocessableEntityException, ValueNotSupportedException
+from slamd.common.session_service import SessionService
+
+session_bp = Blueprint('session', __name__, url_prefix='/session')
+
+
+@session_bp.route('/save', methods=['GET'])
+def save_session():
+    time_string = datetime.now().strftime('%Y-%m-%d_%H%M%S')
+    json_string = SessionService.convert_session_to_json_string()
+    response = make_response(json_string.encode())
+    response.headers['Content-Disposition'] = f'attachment; filename=session_{time_string}.json'
+    response.mimetype = 'text/json'
+    return response
+
+
+@session_bp.route('/restore', methods=['POST'])
+def load_session():
```

Comment:
It was originally called restore_session() since the idea was to delete existing data when loading; when I renamed everything I forgot to change the endpoint. I renamed it to /load for now, but I will take a more thorough look at the entire API as per the other suggestions.